crates/subspace-farmer-components/src/proving.rs

```diff
@@ -140,21 +147,21 @@ where
         self.chunk_candidates.is_empty()
     }
 
-    pub fn into_iter<RewardAddress, PosTable>(
+    pub fn into_solutions<RewardAddress, PosTable>(
         self,
         reward_address: &'a RewardAddress,
         kzg: &'a Kzg,
         erasure_coding: &'a ErasureCoding,
         table_generator: &'a mut PosTable::Generator,
     ) -> Result<
-        impl ExactSizeIterator<Item = Result<Solution<PublicKey, RewardAddress>, ProvingError>> + 'a,
+        impl ProvableSolutions<Item = Result<Solution<PublicKey, RewardAddress>, ProvingError>> + 'a,
```

Comment:
This is a return type, you can't use `where` for it. And because out iterator has a number of generics I wanted to erase its type such that we don't have to propagate those generics any further. Spelling out `SolutionCandidatesIterator` with all generics here will not be much more readable I'm afraid.